types/express-session/index.d.ts

```diff
@@ -9,115 +10,354 @@
+        regenerate(req: express.Request, callback: (err?: any) => any): void;
+        load(sid: string, callback: (err: any, session?: SessionData) => any): void;
+        createSession(req: express.Request, session: SessionData): void;
+
+        /**
+         * Gets the session from the store given a session ID and passes it to `callback`.
+         *
+         * The `session` argument should be a `Session` object if found, otherwise `null` or `undefined` if the session was not found and there was no error.
+         * A special case is made when `error.code === 'ENOENT'` to act like `callback(null, null)`.
+         */
+        abstract get(sid: string, callback: (err: any, session?: SessionData | null) => void): void;
+
+        /** Upsert a session in the store given a session ID and `SessionData` */
+        abstract set(sid: string, session: SessionData, callback?: (err?: any) => void): void;
+
+        /** Destroys the dession with the given session ID. */
+        abstract destroy(sid: string, callback?: (err?: any) => void): void;
+
+        /** Returns all sessions in the store */
+        // https://github.com/DefinitelyTyped/DefinitelyTyped/pull/38783, https://github.com/expressjs/session/pull/700#issuecomment-540855551
+        all?(callback: (err: any, obj?: SessionData[] | { [sid: string]: SessionData; } | null) => void): void;
+
+        /** Returns the amount of sessions in the store. */
+        length?(callback: (err: any, length: number) => void): void;
+
+        /** Delete all sessions from the store. */
+        clear?(callback?: (err?: any) => void): void;
+
+        /** "Touches" a given session, resetting the idle timer. */
+        touch?(sid: string, session: SessionData, callback?: () => void): void;
+    }
 
-declare namespace session {
-  interface SessionOptions {
-    secret: string | string[];
-    name?: string;
-    store?: Store | MemoryStore;
-    cookie?: {
-      maxAge?: number;
-      signed?: boolean;
-      expires?: Date;
-      httpOnly?: boolean;
-      path?: string;
-      domain?: string;
-      secure?: boolean | 'auto';
-      encode?: (val: string) => string;
-      sameSite?: boolean | 'lax' | 'strict' | 'none';
-    };
-    genid?(req: express.Request): string;
-    rolling?: boolean;
-    resave?: boolean;
-    proxy?: boolean;
-    saveUninitialized?: boolean;
     /**
-     * Control the result of unsetting req.session (through delete, setting to null, etc.).
-     * - 'destroy' The session will be destroyed (deleted) when the response ends.
-     * - 'keep' The session in the store will be kept, but modifications made during the request are ignored and not saved.
-     * @default 'keep'
+     * **Warning:** the default server-side session storage, `MemoryStore`, is purposely not designed for a production environment.
+     * It will leak memory under most conditions, does not scale past a single process, and is only meant for debugging and developing.
+     *
+     * @deprecated
```

Comment:
from where this `@deprecated` comes?

Reply:
I honestly do not remember, I'll remove it